File: Lib/Telerik/Examples/QuickStart/GridView/Hierarchy/TabbedChildViews/CustomDetailViewCellElement.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;
using Telerik.Examples.WinControls.Properties;

namespace Telerik.Examples.WinControls.GridView.Hierarchy.TabbedChildViews
{
    public class CustomDetailViewCellElement: GridDetailViewCellElement
    {
        public CustomDetailViewCellElement(GridViewColumn column, GridRowElement row): base(column, row)
        {
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(GridDetailViewCellElement);
            }
        }

        protected override RadPageViewItem CreatePageViewItem(IRadPageViewProvider pageViewProvider, GridViewInfo viewInfo)
        {
            RadPageViewItem item = base.CreatePageViewItem(pageViewProvider, viewInfo);

            RadPageViewExplorerBarItem explorerBarItem = item as RadPageViewExplorerBarItem;
            if (explorerBarItem != null)
            {
                explorerBarItem.AssociatedContentAreaElement.Children[0].MaxSize = new System.Drawing.Size(0, 152);
            }

            return item;
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/GridView/Hierarchy/TabbedChildViews/CustomDetailViewCellElement.vb
Imports System.Text
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.GridView.Hierarchy.TabbedChildViews
	Public Class CustomDetailViewCellElement
		Inherits GridDetailViewCellElement
		Public Sub New(ByVal column As GridViewColumn, ByVal row As GridRowElement)
			MyBase.New(column, row)
		End Sub

		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(GridDetailViewCellElement)
			End Get
		End Property

		Protected Overrides Function CreatePageViewItem(ByVal pageViewProvider As IRadPageViewProvider, ByVal viewInfo As GridViewInfo) As RadPageViewItem
			Dim item As RadPageViewItem = MyBase.CreatePageViewItem(pageViewProvider, viewInfo)

			Dim explorerBarItem As RadPageViewExplorerBarItem = TryCast(item, RadPageViewExplorerBarItem)
			If explorerBarItem IsNot Nothing Then
				explorerBarItem.AssociatedContentAreaElement.Children(0).MaxSize = New Size(0, 152)
			End If

			Return item
		End Function
	End Class
End Namespace
